cluster/examples/kubernetes/edgefs/csi/iscsi/edgefs-iscsi-csi-driver.yaml

```diff
@@ -0,0 +1,338 @@
+# ---------------------------------
+#
+# Runs single driver controller server (driver + provisioner + attacher + snapshotter) on one of the nodes
+#
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: edgefs-iscsi-csi-controller-service-account
+  namespace: default # replace with non-default namespace name if needed
+---
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: edgefs-iscsi-csi-controller-cluster-role
+rules:
+  - apiGroups: ['']
+    resources: ['secrets']
+    verbs: ['get', 'list']
+  - apiGroups: ['']
+    resources: ['persistentvolumes']
+    verbs: ['get', 'list', 'watch', 'create', 'update', 'delete'] # "update" for attacher
+  - apiGroups: ['']
+    resources: ['persistentvolumeclaims']
+    verbs: ['get', 'list', 'watch', 'update']
+  - apiGroups: ['storage.k8s.io']
+    resources: ['storageclasses']
+    verbs: ['get', 'list', 'watch']
+  - apiGroups: ['']
+    resources: ['events']
+    verbs: ['list', 'watch', 'create', 'update', 'patch']
+  - apiGroups: ['snapshot.storage.k8s.io']
+    resources: ['volumesnapshots']
+    verbs: ['get', 'list']
+  - apiGroups: ['snapshot.storage.k8s.io']
+    resources: ['volumesnapshotcontents']
+    verbs: ['get', 'list']
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["get", "list", "watch", "update"]
+  # attacher specific
+  - apiGroups: ['']
+    resources: ['nodes']
+    verbs: ['get', 'list', 'watch']
+  - apiGroups: ['csi.storage.k8s.io']
+    resources: ['csinodeinfos']
+    verbs: ['get', 'list', 'watch']
+  - apiGroups: ['storage.k8s.io']
+    resources: ['volumeattachments']
+    verbs: ['get', 'list', 'watch', 'update']
+  # snapshotter specific
+  - apiGroups: ['snapshot.storage.k8s.io']
+    resources: ['volumesnapshotclasses']
+    verbs: ['get', 'list', 'watch']
+  - apiGroups: ['snapshot.storage.k8s.io']
+    resources: ['volumesnapshotcontents']
+    verbs: ['create', 'get', 'list', 'watch', 'update', 'delete']
+  - apiGroups: ['snapshot.storage.k8s.io']
+    resources: ['volumesnapshots']
+    verbs: ['get', 'list', 'watch', 'update']
+  - apiGroups: ['apiextensions.k8s.io']
+    resources: ['customresourcedefinitions']
+    verbs: ['create', 'list', 'watch', 'delete']
+---
+
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: edgefs-iscsi-csi-controller-cluster-role-binding
+subjects:
+  - kind: ServiceAccount
+    name: edgefs-iscsi-csi-controller-service-account
+    namespace: default # replace with non-default namespace name if needed
+roleRef:
+  kind: ClusterRole
+  name: edgefs-iscsi-csi-controller-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+
+kind: Service
+apiVersion: v1
+metadata:
+  name: edgefs-iscsi-csi-controller
+  labels:
+    app: edgefs-iscsi-csi-controller
+spec:
+  selector:
+    app: edgefs-iscsi-csi-controller
+  ports:
+    - name: dummy
+      port: 12345
+---
+
+kind: StatefulSet
+apiVersion: apps/v1beta1
```

Comment:
Update the DaemonSet to `apps/v1`, for that a `selector` section must be added to the DaemonSet Spec.

Reply:
Nope, https://github.com/sabbot/rook/blob/master/cluster/examples/kubernetes/edgefs/csi/iscsi/edgefs-iscsi-csi-driver.yaml#L127-L138